File: legacy/src/GitOpen.Application/Workspaces/Workspace.cs
using GitOpen.Domain.Commits;
using GitOpen.Domain.Repositories;

namespace GitOpen.Application.Workspaces;

public sealed class Workspace
{
    public RepoLocation Location { get; }
    public string? SelectedBranchFullName { get; set; }
    public CommitSha? SelectedSha { get; set; }
    public int ScrollOffset { get; set; }

    public Workspace(RepoLocation location) => Location = location;
}

File: lib/application/workspaces/workspace.dart
import 'package:gitopen/domain/commits/commit_sha.dart';
import 'package:gitopen/domain/repositories/repo_location.dart';

final class Workspace {

  Workspace(this.location, {
    this.selectedBranchFullName,
    this.selectedSha,
    this.scrollOffset = 0,
  });
  final RepoLocation location;
  String? selectedBranchFullName;
  CommitSha? selectedSha;
  int scrollOffset;
}
